libethereum/EthereumHost.cpp

```diff
@@ -576,8 +576,20 @@ void EthereumHost::onPeerNewBlock(EthereumPeer* _peer, RLP const& _r)
 				u256 difficulty = _r[1].toInt<u256>();
 				if (m_syncingTotalDifficulty < difficulty)
 				{
+					BlockInfo bi;
+					RLP root(_r[0].data());
+					try
+					{
+						bi.populateFromHeader(root[0], CheckEverything, h);
```

Comment:
An attacker can pregenerate any number of blocks with valid nonce and minimum difficulty and feed them here. BlockQueue now has a limit of unknown blocks it can handle that is supposed to protect against this.